File: src/main/java/com/otus/service/UserServiceImpl.java
package com.otus.service;

import lombok.Data;

import java.util.Scanner;

import static com.otus.utils.PrintUtils.print;

@Data
public class UserServiceImpl implements UserService {

    private String firstName;
    private String lastName;
    private static final Scanner scanner = new Scanner(System.in);

    @Override
    public void askFirstName() {
        print("Your first name: ");
        firstName = scanner.next();
    }

    @Override
    public void askLastName() {
        print("Your last name: ");
        lastName = scanner.next();
    }

    @Override
    public void showFullName() {
        print(firstName + " " + lastName);
    }
}

File: src/main/java/com/otus/service/UserServiceImpl.kt
package com.otus.service

import com.otus.service.localization.LocalizationLabels.ASK_FIRST_NAME
import com.otus.service.localization.LocalizationLabels.ASK_LAST_NAME
import com.otus.service.localization.LocalizationService
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(
    private val localizationService: LocalizationService,
    private val scanner: ConsoleInputService,
    private val printService: PrintService
) : UserService {

    private lateinit var firstName: String
    private lateinit var lastName: String

    override fun askFirstName() {
        val demandFirstName = localizationService.getLocalizedStringByLabel(ASK_FIRST_NAME)
        printService.println(demandFirstName)
        firstName = scanner.next() ?: ""
    }

    override fun askLastName() {
        val demandLastName = localizationService.getLocalizedStringByLabel(ASK_LAST_NAME)
        printService.println(demandLastName)
        lastName = scanner.next() ?: ""
    }

    override fun showFullName() {
        printService.println("$firstName $lastName")
    }
}